### PR TITLE
get_clip_path checks for nan

### PR DESCRIPTION
**Pull request by leejjoon**
## PR Summary

Specifying an ill-defined *clip_path* leads to a seg-fault (#4448) with agg backends. Here is an minimal example.

```python
import io
import numpy as np

import matplotlib.figure as mfigure
from matplotlib.projections import PolarAxes
import matplotlib.patches as mpatches
from matplotlib.backends.backend_agg import FigureCanvasAgg

fig = mfigure.Figure()
ax = fig.add_subplot(111)
l1, = ax.plot([-1, 1], [-1, 1])

tr = PolarAxes.PolarTransform()
p = mpatches.Polygon(np.arange(-3, 3).reshape((3, 2)), transform=tr)

l1.set_clip_path(p)

FigureCanvasAgg(fig).print_png(io.BytesIO())

```

The current PR simply fix this by making *GraphicsContextBase.get_clip_path* returns None if it contains values that are not finite.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by anntzer** [comment]
I would rather prefer we detect the invalid path and error out appropriately...

File: lib/matplotlib/backend_bases.py [hunk 1]
```diff
@@ -808,7 +808,12 @@ def get_clip_path(self):
         an affine transform to apply to the path before clipping.
         """
         if self._clippath is not None:
-            return self._clippath.get_transformed_path_and_affine()
+            tpath, tr = self._clippath.get_transformed_path_and_affine()
+            if np.all(np.isfinite(tpath.vertices)):
+                return tpath, tr
+            else:
+                warnings.warn("Ill-defined clip_path detected. Returning None.")
```
Comment (anntzer) [inline lib/matplotlib/backend_bases.py]: This should be _log.warning(...).

**Comment by anntzer** [comment]
Actually, with the warning I'm fine with this approach.  Good to go modulo one comment.

File: lib/matplotlib/backend_bases.py (hunk 1 above)
Comment (timhoffm) [reply lib/matplotlib/backend_bases.py]: Which will also reduce the line length of that line by 1 and thus make Travis (flake8) pass.

Comment (jklymak) [inline lib/matplotlib/backend_bases.py:815]:
```suggestion
                _log.warning("Ill-defined clip_path detected. Returning None.")
```

**Comment by jklymak** [comment]
@leejjoon I made the change requested above and rebased on current master and pushed this to your branch.  Hope that is OK

**Review by jklymak — APPROVED** [review]

**Pull request merged**